### ISSUE TITLE
[Bug]: Different responses from two queries with identical evaluated value

### ISSUE THREAD

**Issue by micahgrinnell** [appsmithorg/appsmith#11288]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Here is my SQL query that is causing me trouble. It pulls items from an inventory Postgres database that match the MFGPartNumber and Condition Value. 
```
SELECT * 
FROM public."formatted_list" 
WHERE "MFGPartNumber" ILIKE '{{searchBar.text}}' 
AND "Condition" IN ({{conditionSelect.selectedOptionValue}})
```
Here is its evaluated value:
```
SELECT * 
FROM public."formatted_list" 
WHERE "MFGPartNumber" ILIKE '003-1436-001' 
AND "Condition" IN ('R','N','Z')
```
**When run, nothing is returned:**
![appsmith_noresponse](https://user-images.githubusercontent.com/47792060/154709305-ab95f91c-6cca-45d6-9e86-8eabc03f3d98.PNG)

**Yet, when I replace ({{conditionSelect.selectedOptionValue}}) with ('R','N','Z') I get a response with an accurate matching item:**

![appsmith_response](https://user-images.githubusercontent.com/47792060/154707824-0548ada0-c76e-4729-a67b-691a4ea8022e.PNG)

Here is the query that gets a response (identical to the evaluated value of the first query shown):
```
SELECT * 
FROM public."formatted_list" 
WHERE "MFGPartNumber" ILIKE '003-1436-001' 
AND "Condition" IN ('R','N','Z')
```
Here are the options in the conditionSelect widget:
```
[
{
    "label": "Any",
    "value": "'R','N','Z'"
  },
  {
    "label": "Refurbished",
    "value": "'R'"
  },
  {
    "label": "New",
    "value": "'N'"
  }
]
``` 
Is this a problem with assigning multiple strings within the value of the "Any" option? Or is there something more going on here?

I'd be happy to provide more context if required, I'm new to this bug report process so forgive me if this is lacking in detail. 

### Steps To Reproduce

1. Request table data from Postgres server connection
2. Integrate multi-option widget with a value of separated chars within one string ("'R','Z','N'" as per my config)
3. Create query that searches through Postgres table based on the values set in the multi-option widget. 
4. Profit??

### Environment

Production

### Version

Cloud

**Comment by micahgrinnell** [comment]
I tried turning off Prepared Statements as I saw in https://github.com/appsmithorg/appsmith/issues/11259. Now the code runs, but I would still like to get this resolved.

**Comment by Nikhil-Nandagopal** [comment]
@micahgrinnell agreed we should solve this. I'm closing this issue as this will be solved in #11259
